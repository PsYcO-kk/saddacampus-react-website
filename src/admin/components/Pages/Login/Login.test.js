import React from 'react';
import {shallow} from 'enzyme';
import { Login } from './Login';

let wrapper;

beforeEach(() => {
	wrapper = shallow(<Login />);
})

test('should render Login component without crashing', () => {
	expect(wrapper).toMatchSnapshot();
});

test('should disable error when username or password is changed', () => {
	wrapper.setState({ error: 'xyz' });
	wrapper.find('Input[type="text"]').simulate('change');
	expect(wrapper.state('error')).toBe('');
});
