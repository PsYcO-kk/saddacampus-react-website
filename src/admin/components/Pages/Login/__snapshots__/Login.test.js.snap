// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login component without crashing 1`] = `
<div
  className="app flex-row align-items-center"
>
  <Container
    tag="div"
  >
    <Row
      className="justify-content-center"
      tag="div"
    >
      <Col
        md="8"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <CardGroup
          tag="div"
        >
          <Card
            className="p-4"
            tag="div"
          >
            <CardBody
              tag="div"
            >
              <form
                onSubmit={[Function]}
              >
                <h1>
                  Login
                </h1>
                <p
                  className="text-muted"
                >
                  Sign In to your account
                </p>
                <InputGroup
                  className="mb-3"
                  tag="div"
                >
                  <InputGroupAddon
                    addonType="prepend"
                    tag="div"
                  >
                    <InputGroupText
                      tag="span"
                    >
                      <i
                        className="icon-user"
                      />
                    </InputGroupText>
                  </InputGroupAddon>
                  <Input
                    name="username"
                    onChange={[Function]}
                    placeholder="Username"
                    required={true}
                    type="text"
                  />
                </InputGroup>
                <InputGroup
                  className="mb-4"
                  tag="div"
                >
                  <InputGroupAddon
                    addonType="prepend"
                    tag="div"
                  >
                    <InputGroupText
                      tag="span"
                    >
                      <i
                        className="icon-lock"
                      />
                    </InputGroupText>
                  </InputGroupAddon>
                  <Input
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    required={true}
                    type="password"
                  />
                </InputGroup>
                <Row
                  tag="div"
                >
                  <Col
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                    xs="6"
                  >
                    <Button
                      className="px-4"
                      color="primary"
                      tag="button"
                      type="submit"
                    >
                      Login
                    </Button>
                  </Col>
                  <Col
                    className="text-right"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                    xs="6"
                  >
                    <Button
                      className="px-0"
                      color="link"
                      tag="button"
                      type="button"
                    >
                      Forgot password?
                    </Button>
                  </Col>
                </Row>
              </form>
            </CardBody>
          </Card>
        </CardGroup>
      </Col>
    </Row>
  </Container>
</div>
`;
